public/app/features/dashboard-scene/scene/VariableControls.tsx

```diff
@@ -15,18 +15,45 @@ import {
 import { useElementSelection, useStyles2 } from '@grafana/ui';
 
 import { DashboardScene } from './DashboardScene';
+import { DrilldownControls } from './DrilldownControls';
 import { AddVariableButton } from './VariableControlsAddButton';
 
 export function VariableControls({ dashboard }: { dashboard: DashboardScene }) {
   const { variables } = sceneGraph.getVariables(dashboard)!.useState();
+  const styles = useStyles2(getStyles);
+
+  // Get visible variables for drilldown layout
+  const visibleVariables = variables.filter((v) => v.state.hide !== VariableHide.hideVariable);
+
+  const adHocVar = visibleVariables.find((v) => sceneUtils.isAdHocVariable(v));
+  const groupByVar = visibleVariables.find((v) => sceneUtils.isGroupByVariable(v));
+
+  const showDrilldownControls = config.featureToggles.dashboardAdHocAndGroupByWrapper && adHocVar && groupByVar;
+
+  const restVariables = visibleVariables.filter(
+    (v) => v.state.name !== adHocVar?.state.name && v.state.name !== groupByVar?.state.name
+  );
+
+  // Variables to render after the drilldown row (or all variables if not using drilldown layout)
+  const variablesToRender = showDrilldownControls
+    ? restVariables.filter((v) => v.state.hide !== VariableHide.inControlsMenu)
+    : variables.filter((v) => v.state.hide !== VariableHide.inControlsMenu);
 
   return (
     <>
-      {variables
-        .filter((v) => v.state.hide !== VariableHide.inControlsMenu)
-        .map((variable) => (
-          <VariableValueSelectWrapper key={variable.state.key} variable={variable} />
-        ))}
+      {showDrilldownControls && (
+        <div className={styles.drilldownControlsWrapper}>
+          <DrilldownControls adHocVar={adHocVar} groupByVar={groupByVar} />
+        </div>
+      )}
```

Comment:
I think a solution to achieve the designs as @Ijin08 suggests is to move these DrilldownControls up in DashboardControls instead of keeping them here. We should just exclude them from here and render the remainder of the variables as before, while the groupBy and adHoc live on the same row as timerange or refresh picker and float properly there